### ISSUE TITLE
Test problems

### ISSUE THREAD

**Issue by russel** [DmitryOlshansky/dinotify#3]
Running on Debian Sid fully up to date as of 2017-11-06, using DMD from D-Apt:

  DMD64 D Compiler v2.077.0

and ldc2 from Debian packages:

  LDC - the LLVM D compiler (1.4.0):
    based on DMD v2.074.1 and LLVM 5.0.0
    built with LDC - the LLVM D compiler (0.17.5)
    Default target: x86_64-pc-linux-gnu

with dub from D-apt:

  DUB version 1.6.0, built on Nov  1 2017

Running the tests with dmd, there appears to be a deadlock, or the tests take a very long time:
```
|> dub test
Generating test runner configuration 'dinotify-test-library' for 'library' (library).
Performing "unittest" build using dmd for x86_64.
dinotify 0.2.1+commit.1.g941e980: building configuration "dinotify-test-library"...
Linking...
Running ./dinotify-test-library 
^C
```

Running the tests with ldc2 leads to the same problem:
```
|> dub test --compiler=ldc2
Generating test runner configuration 'dinotify-test-library' for 'library' (library).
Performing "unittest" build using ldc2 for x86_64.
dinotify 0.2.1+commit.1.g941e980: building configuration "dinotify-test-library"...
Running ./dinotify-test-library 
^C
```

**Comment by DmitryOlshansky** [comment]
It’s been a while since I run test suite but I think it may have to do with linux headers change in recent DMD.

Could you try an earlier version(s) to see if it works?

I’ll try it out as well but no sonner then tomorrow.

**Comment by russel** [comment]
No worries. I am creating a Meson build for this to create a shared library rather than take a vendor copy of the source (*). I shall see if it works or if there really is a problem.

(*) I am doing a GtkD program and so shared libraries of dependencies are essential, dub is almost completely useless for this, especially as I have a special build of GtkD.

**Comment by DmitryOlshansky** [comment]
@russel Should be fixed now, check out v0.2.2 or current master.

**Comment by russel** [comment]
Tests all pass using the Meson build with Ninja :-) Also using Dub :-) :-)
